examples/http-rust-outbound-http/outbound-http-to-same-app/src/lib.rs

```diff
@@ -0,0 +1,47 @@
+use anyhow::Result;
+use spin_sdk::{
+    http::{Request, Response},
+    http_component,
+};
+use url::{Position, Url};
+
+const COMPONENT_ROUTE: &str = "/hello";
+
+/// Send an HTTP request and return the response.
+#[http_component]
+fn send_outbound(req: Request) -> Result<Response> {
+    let outbound_uri = base_url(req)?.join(COMPONENT_ROUTE).unwrap();
+
+    let mut res = spin_sdk::outbound_http::send_request(
+        http::Request::builder()
+            .method("GET")
+            .uri(outbound_uri.as_str())
+            .body(None)?,
+    )?;
+    res.headers_mut()
+        .insert("spin-component", "rust-outbound-http".try_into()?);
+    println!("{:?}", res);
+    Ok(res)
+}
+
+fn base_url(req: Request) -> Result<Url> {
```

Comment:
Is it an option to provide a function like this in the SDKs, so users don't have to implement this in every component?

Reply:
Alternatively we could reserve some fake host name to represent "self", e.g. `https://self.local`.

Reply:
`https://my.ego` :-) Demo time "So here I call my ego" 👍🏻